### PR TITLE
Checkout for user, closes #22

### LINKED ISSUE #22: 15: User can check out
Background: An existing user and a cart with items

As a visitor 
When I add items to my cart 
And I visit "/cart" 
And I click "Login or Register to Checkout" 
Then I should be required to login

When I am logged in 
And I visit my cart 
And when I click "Checkout" 
Then the order should be placed 
And my current page should be "/orders" 
And I should see a message "Order was successfully placed" 
And I should see the order I just placed in a table

### PR DESCRIPTION
**Pull request by MikelSage**
Adds ability to checkout for user.

### CODE REVIEW

File: app/controllers/orders_controller.rb [hunk 1]
```diff
@@ -12,4 +12,14 @@ def index
...
     @order = Order.find(params[:id])
   end
+
+  def create
+    order = Order.create(user_id: current_user.id, order_items_attributes: @cart.order_items_attributes)
+
+    flash[:success] = 'Order was successfully placed'
+    @cart.contents.clear
+    session[:cart].clear if session[:cart]
+
```
Comment (swdonovan) [inline app/controllers/orders_controller.rb:22]: As discussed, this is either brilliant or over guard....we'll figure it out lol

File: app/models/cart.rb [hunk 2]
```diff
@@ -30,4 +30,15 @@ def increase_item(item_id)
...
+
+  def order_items_attributes
+    @contents.map do |id, quantity|
+      price = Item.find(id).price
+      {
+        item_id: id.to_i,
+        item_quantity: quantity,
+        line_item_total: quantity * price
+      }
+    end
```
Comment (swdonovan) [inline app/models/cart.rb:42]: This is absolutely beautiful and makes the program so much more flexible and growable. Nice work dude!

File: app/views/carts/index.html.erb [hunk 3]
```diff
@@ -30,7 +30,7 @@
     <div class="col-md-2 pull-right">
       <p>Total Price: $<%= @cart.total_price %></p>
       <% if current_user %>
-        <%= link_to 'Checkout', '#', class: 'btn btn-primary' %>
+        <%= link_to 'Checkout', orders_path, method: :post, class: 'btn btn-primary' %>
       <% else %>
```
Comment (jimszalew) [inline app/views/carts/index.html.erb:34]: nice work here

File: app/controllers/orders_controller.rb (hunk 1 above)
Comment (jimszalew) [reply app/controllers/orders_controller.rb:22]: I concur

**Review by jimszalew — APPROVED** [review]
scoot master is wise

**Pull request merged**